standalone-metastore/metastore-server/src/test/java/org/apache/hadoop/hive/metastore/client/TestTablesCreateDropAlterTruncate.java

```diff
@@ -1567,6 +1572,24 @@ public void dropTableBogusCatalog() throws TException {
     client.dropTable("nosuch", testTables[0].getDbName(), testTables[0].getTableName(), true, false);
   }
 
+  @Test
+  public void testDropExternalTableWithoutStoragePermission() throws TException, IOException {
```

Comment:
add a test case where data was supposed to be deleted & the user didn't had permission & assert that the exception is thrown